src/main/java/spoon/reflect/declaration/CtAnnotation.java

```diff
@@ -165,4 +167,42 @@
 	@Override
 	@UnsettableProperty
 	<C extends CtExpression<A>> C setTypeCasts(List<CtTypeReference<?>> types);
+
+	static CtAnnotatedElementType getAnnotatedElementTypeForCtElement(CtElement element) {
```

Comment:
use an anonymous scanner to avoid multiple instanceof?

Reply:
I don't see how to use easily an anonymous scanner here: I must return a variable and I cannot store it as a local variable if I use the anonymous scanner. And I don't have any method to call to setup the value.

Reply:
I agree, it's not straightforward.

> I must return a variable and I cannot store it as a local variable if I use the anonymous scanner
 
Unless you create an inline class

```
class Result { CtVariable v }
final Result result = new Result();
// then the anonymous class

```